cot/src/admin.rs

```diff
@@ -144,16 +171,44 @@ async fn view_model(request: Request) -> cot::Result<Response> {
         model: &'a dyn AdminModelManager,
         #[debug("..")]
         objects: Vec<Box<dyn AdminModel>>,
+        page: usize,
+        page_size: &'a usize,
+        total_pages: usize,
     }
 
     let model_name: String = request.path_params().parse()?;
     let manager = get_manager(&request, &model_name)?;
 
+    let query_params: HashMap<String, String> = request
+        .uri()
+        .query()
+        .map(|q| {
+            query_pairs(&Bytes::copy_from_slice(q.as_bytes()))
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect()
+        })
+        .unwrap_or_default();
+
+    let page: usize = query_params
+        .get("page")
+        .map_or(1, |p| p.parse().unwrap_or(1));
+    let limit = query_params
+        .get("page_size")
+        .map_or(10, |p| p.parse().unwrap_or(10));
+    let pagination = Pagination::new(limit, page);
+    let objects = manager.get_objects(&request, pagination).await?;
+    let total_objects = manager.get_total_object_counts(&request).await?;
+    let total_pages = total_objects.div_ceil(limit);
```

Comment:
Maybe we should throw a 404 error when users tries to reach page 0 or a page higher than the total number of pages? Right now we just return 500 Internal Server Error in former case and 0 records in the latter case and I think none of these is good for the user.